compiler/grainformat/reformat.re

```diff
@@ -460,23 +461,39 @@ let rec block_item_iterator_line =
 
       let comments_without_item_comments =
         remove_used_comments(
-          ~all_comments=comments_without_leading,
-          item_comments,
+          ~remove_comments=item_comments,
+          comments_without_leading,
         );
 
-      Doc.concat([
-        this_item,
-        block_item_iterator_line(
-          ~previous=PreviousItem(get_loc(item)),
-          ~get_loc,
-          ~print_item,
-          ~comments=comments_without_item_comments,
-          ~get_attribute_text,
-          ~original_source,
-          ~separator,
-          remainder,
-        ),
-      ]);
+      switch (separator) {
+      | None =>
+        Doc.concat([
+          this_item,
+          block_item_iterator_line(
+            ~previous=PreviousItem(get_loc(item)),
+            ~get_loc,
+            ~print_item,
+            ~comments=comments_without_item_comments,
+            ~get_attribute_text,
+            ~original_source,
+            remainder,
+          ),
+        ])
+      | Some(sep) =>
+        Doc.concat([
+          this_item,
+          block_item_iterator_line(
+            ~previous=PreviousItem(get_loc(item)),
+            ~get_loc,
+            ~print_item,
+            ~comments=comments_without_item_comments,
+            ~get_attribute_text,
+            ~original_source,
+            ~separator=sep,
+            remainder,
+          ),
+        ])
```

Comment:
These are the same code except for `~separator` right? You can just use `~separator?` to pun and unwrap the optional into the function call.

Reply:
Oooooh. So that's how you do it!

Reply:
Fixed